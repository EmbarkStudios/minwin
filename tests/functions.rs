mod utils;
use utils::*;

// Ensures we can find and emit arch specific functions
test!(
    arches,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            linking_style: LinkingStyle::RawDylib,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.System.Search.SQLBindCol",
        "Windows.Win32.UI.WindowsAndMessaging.GetWindowLongPtrA",
        "Windows.Win32.UI.WindowsAndMessaging.GetWindowLongPtrW",
    ]
);
