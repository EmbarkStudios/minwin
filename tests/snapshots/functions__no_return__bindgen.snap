---
source: tests/functions.rs
expression: bo.bindings
---
// Bindings generated by `windows-bindgen` 0.49.0

#![allow(
    non_snake_case,
    non_upper_case_globals,
    non_camel_case_types,
    dead_code,
    clippy::all
)]
::windows_targets::link!("kernel32.dll" "system" fn ExitProcess(uexitcode : u32) -> !);
::windows_targets::link!("kernel32.dll" "system" fn ExitThread(dwexitcode : u32) -> !);
::windows_targets::link!("kernel32.dll" "system" fn FatalExit(exitcode : i32) -> !);
::windows_targets::link!("kernel32.dll" "system" fn FreeLibraryAndExitThread(hlibmodule : HMODULE, dwexitcode : u32) -> !);
pub type HMODULE = isize;

